src/scipp/core/dimensions.py

```diff
@@ -1,12 +1,57 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
 # @author Simon Heybrock
-from typing import Dict
+from typing import Dict, Optional
 
 from .._scipp.core import Variable, DataArray, Dataset, CoordError
+from ..typing import VariableLikeType
 from .dataset import merge
 
 
+def _combine_dims(dims_dict: Optional[Dict[str, str]],
+                  names: Dict[str, str]) -> Dict[str, str]:
+    dims_dict = {} if dims_dict is None else dims_dict
+    if set(dims_dict).intersection(names):
+        raise ValueError(
+            'The names passed in the dict and as keyword arguments must be distinct.'
+            f'Got dims_dict={dims_dict} and kwargs={names}')
```

Comment:
`dims_dict` is position-only, should its name be mentioned?

Reply:
It still shows up in the docs. But I can change the message.

Reply:
I don't know what the best practice is. I find it odd that pos-only args are named in the docs (but that is not something we can change).

Reply:
Since it is named in the arg list of the docs, keeping the name here makes sense.